plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java

```diff
@@ -290,11 +290,16 @@ private void importHierarchicalGraph(final ElkNode elkgraph, final LGraph lgraph
         
         Direction parentGraphDirection = lgraph.getProperty(LayeredOptions.DIRECTION);
 
+        // Model order index for nodes
+        int index = 0;
         // Transform the node's children
         elkGraphQueue.addAll(elkgraph.getChildren());
         while (!elkGraphQueue.isEmpty()) {
             ElkNode elknode = elkGraphQueue.poll();
             
+            // Assign a model order to the edges as they are read
+            elknode.setProperty(InternalProperties.MODEL_ORDER, index++);
```

Comment:
I can't remember what we decided on the last time but maybe only set the property if any option is activated that relies on the model order?

Reply:
I think that sounds reasonable. I will add that to the "normal" case too.